File: v1/src/main/java/com/api/v1/car/mappers/CarResponseMapper.java
package com.api.v1.car.mappers;

import com.api.v1.car.domain.Car;
import com.api.v1.car.dtos.CarResponseDto;

public record CarResponseMapper(Car car) {

    public CarResponseDto mapToDto() {
        return new CarResponseDto(
                car.getMake(),
                car.getModel(),
                car.getProductionYear(),
                car.getVin(),
                car.getProductionYear(),
                car.getAddedAt(),
                car.getUpdatedAt()
        );
    }

}

File: kotlin/v2/src/main/kotlin/com/api/v2/car/mappers/CarResponseMapper.kt
package com.api.v2.car.mappers

import com.api.v2.car.domain.Car
import com.api.v2.car.dtos.CarResponseDto

class CarResponseMapper {

    companion object {
        fun mapToDto(car: Car): CarResponseDto {
            return CarResponseDto(
                car.vin,
                car.model,
                car.productionYear,
                car.make,
                car.plateNumber,
                car.createdAt,
                car.updatedAt
            )
        }
    }

}